src/btoropt.c

```diff
@@ -110,6 +111,21 @@ strcmpoptval (const char *a, const char *b)
   return strncmp (a, b, len_a);
 }
 
+static void
+btor_opt_check_solver_supports_timeout (uint32_t timeout, uint32_t sat_engine)
+{
+  if (timeout
+      && (sat_engine == BTOR_SAT_ENGINE_CMS
+          || sat_engine == BTOR_SAT_ENGINE_MINISAT
+          || sat_engine == BTOR_SAT_ENGINE_PICOSAT))
+  {
+    BTOR_WARN (true,
+               "SAT solver %s does not support termination functions; timeout "
+               "option may not have effect",
```

Comment:
This check could be done in [btor_sat_init](https://github.com/Boolector/boolector/blob/master/src/btorsat.c#L355). It should be enough to check if the timeout option is enabled and if `smgr->api.setterm` is available.